sdk/samples/iot/provisioning/CMakeLists.txt

```diff
@@ -29,4 +29,16 @@ target_link_libraries(paho_iot_provisioning_sas_example
     az::iot::samples::common
 )
 
+# Workaround for linker warning LNK4098: defaultlib 'LIBCMTD' conflicts with use of other libs
+if (MSVC)
+    set_target_properties(paho_iot_provisioning_example
+        PROPERTIES LINK_FLAGS
+        "/NODEFAULTLIB:libcmtd.lib"
+    )
+    set_target_properties(paho_iot_provisioning_sas_example
+        PROPERTIES LINK_FLAGS
+        "/NODEFAULTLIB:libcmtd.lib"
```

Comment:
you can do `set_target_propiertes(target1 target2...)`  instead of repeating the command
https://cmake.org/cmake/help/latest/command/set_target_properties.html

Reply:
another approach would be trying to use `set_property(GLOBAL)` to affect all targets in the project with this property LINK_FLAGS
see: https://cmake.org/cmake/help/latest/command/set_property.html#command:set_property

Reply:
ah, and it looks like `LINK_FLAGS` should be avoided and replaced by `LINK_OPTIONS`
see https://cmake.org/cmake/help/latest/prop_tgt/LINK_FLAGS.html#prop_tgt:LINK_FLAGS

Reply:
There is also eng/cmake/global_compile_options.txt for the things that should affect all projects

Reply:
Passing that option to `global_compile_options` isn't working (option is unknown). I get the following warning:
> cl : command line warning D9002: ignoring unknown option '/NODEFAULTLIB:libcmtd.lib' [E:\GitHub\Fork\azure-sdk-for-c\build\sdk\samples\storage\blobs\blobs_client_example.vcxproj]

Reply:
> another approach would be trying to use `set_property(GLOBAL)` to affect all targets in the project with this property LINK_FLAGS

That option doesn't work either. I tried adding them to the root cmakelist file, as follows:
```text
# Workaround for linker warning LNK4098: defaultlib 'LIBCMTD' conflicts with use of other libs
if (MSVC)
    set_property(GLOBAL PROPERTY "LINK_FLAGS" "/NODEFAULTLIB:libcmtd.lib")
endif()
```

Any other suggestions?